tests/src/script_tests/withdrawal.rs

```diff
@@ -3,23 +3,494 @@ use super::utils::layer1::build_simple_tx_with_out_point;
+    let expected_err = ScriptError::ValidationFailure(
+        format!(
+            "by-type-hash/{}",
+            ckb_types::H256(script_ctx.withdrawal.script.hash())
+        ),
+        OWNER_CELL_NOT_FOUND_ERROR,
+    )
+    .input_lock_script(0);
+    assert_error_eq!(err, expected_err);
+
+    // Fill correct output
+    let tx = tx
+        .as_advanced_builder()
+        .output(output_cell.0)
+        .output_data(output_cell.1.to_ckb())
+        .build();
+
+    verify_ctx.verify_tx(tx).expect("success");
+}
+
+#[test]
+fn test_unlock_withdrawal_via_finalize_fallback_to_input_owner_cell() {
+    init_env_log();
+
+    const DEFAULT_CAPACITY: u64 = 1000 * 10u64.pow(8);
+
+    let rollup_type_script = random_always_success_script();
+    let rollup_type_hash = rollup_type_script.hash();
+    let (mut verify_ctx, script_ctx) = build_verify_context();
+
+    let last_finalized_block_number = rand::random::<u64>() + 100;
+    let rollup_cell = {
+        let global_state = GlobalState::new_builder()
+            .last_finalized_block_number(last_finalized_block_number.pack())
+            .build();
+
+        let output = CellOutput::new_builder()
+            .lock(random_always_success_script())
+            .type_(Some(rollup_type_script).pack())
+            .capacity(DEFAULT_CAPACITY.pack())
+            .build();
+
+        (output, global_state.as_bytes())
+    };
+    let rollup_dep = {
+        let out_point = verify_ctx.insert_cell(rollup_cell.0.to_ckb(), rollup_cell.1);
+        CellDep::new_builder().out_point(out_point.to_gw()).build()
+    };
+
+    let (sk, pk) = {
+        let secp = Secp256k1::new();
+        let mut rng = OsRng::new().unwrap();
+        secp.generate_keypair(&mut rng)
+    };
+    let owner_lock = {
+        let args = {
+            let mut buf = [0u8; 32];
+            let mut hasher = new_blake2b();
+            hasher.update(&pk.serialize());
+            hasher.finalize(&mut buf);
+
+            Bytes::copy_from_slice(&buf[..20])
+        };
+
+        Script::new_builder()
+            .code_hash(script_ctx.acp.script.hash().pack())
+            .hash_type(ScriptHashType::Type.into())
+            .args(args.pack())
+            .build()
+    };
+    let finalized_withdrawal_cell = {
+        let lock_args = WithdrawalLockArgs::new_builder()
+            .account_script_hash(random_always_success_script().hash().pack())
+            .withdrawal_block_hash(random_always_success_script().hash().pack())
+            .withdrawal_block_number(last_finalized_block_number.saturating_sub(1).pack())
+            .sell_amount(0u128.pack())
+            .sell_capacity(0u64.pack())
+            .owner_lock_hash(owner_lock.hash().pack())
+            .payment_lock_hash(owner_lock.hash().pack())
+            .build();
+
+        let mut args = Vec::new();
+        args.extend_from_slice(&lock_args.as_bytes());
+        assert!(owner_lock.as_bytes().len() <= u32::MAX as usize);
```

Comment:
Why we need this `assert`?

Reply:
Oh, it just help me write test, nothing more.